src/styles/Insights.styles.scss

```diff
@@ -1,4 +1,6 @@
 .insights-page {
+    width: 100%;
+    max-width: 1085px;
```

Comment:
I get why we add a `width` here but why do we add a `max-width`

Reply:
I just realized that also adding width has the chart's grey background filling the whole width
![Capture](https://github.com/egytech-fyi/egytech-fyi/assets/26578518/57102e15-4c5b-434d-b958-06215114a32b)

Reply:
The point of these two lines is to prevent the layout shift caused by the absence of the content.
https://github.com/egytech-fyi/egytech-fyi/assets/112858669/dc83b6a1-605b-4fee-909b-deb79b09106e

Reply:
I get what you mean, how about we do this and let me know what you think

```
.insights-page {
  width: 70%;
  padding: 2.5rem 0;
}

.spinner {
  margin: auto;
}
```
and we add the `spinner` class to the Spinner component like this
`<Suspense fallback={<Spin className='spinner' size='large' />}>`

Reply:
Setting the width to 70% will lead to some issues on smaller screens. I want the content to take the fullscreen width till a certain point, which is at 1085px.

![Screenshot 2024-06-18 185324](https://github.com/egytech-fyi/egytech-fyi/assets/112858669/0c8599bd-6eff-4757-af11-fd31ca12ad26)
![Screenshot 2024-06-18 185347](https://github.com/egytech-fyi/egytech-fyi/assets/112858669/fd29913f-2554-416d-a769-fdb9d05c9c53)

Reply:
Hmmm good point, I guess the thing that this would change is the grey background spanning to 100% of the width which isn't how it looks like now

Reply:
Can you explain more about how this is different from the current version? because I see no difference between this and the current version on production.

Reply:
sorry guys I don't have time due to work at the moment to hop on this conv but I have a couple of pointers if they help

1. I agree, a page shouldn't have a max-width, rather its container should define what upper limit width that page CAN span
2. we have to be mindful of other components having a max width, maybe even rethink how elements should behave and control them from their parents not from within themselves


hope these can help, if not I will try to show you what I mean as soon as I have time to work on this

Reply:
Hi guys, I'm thinking of another approach for this. We can lazy load the entire page instead of its content.
Here's what I did:

```
// App.tsx
const Insights = lazy(() => import('./pages/Insights'))

....

function App() {
 ....
  return (
    <Router>
            .....
            <div className='app-container'>
              <Switch>
                .....
                <Route
                  path='/report'
                  element={
                    <Suspense fallback={<Spin size='large' style={{ margin: 'auto' }} />}>
                      <Insights drawerOpen={visible} onDrawerClose={handleCloseDrawer} />
                    </Suspense>
                  }
                />
                .......
    </Router>
  )
}

export default App
```

```
// insights.styles.scss
.Insights-page {
    min-height: calc(100vh - 64px); // added min-height for the page
    padding: 2.5rem 0;
}
```

```
// Footer.styles
.footers {
    ......
    margin-top: auto; // to keep the footer stick to the bottom of the page
}
```

and that will look like this:
![Screenshot 2024-06-20 065143](https://github.com/egytech-fyi/egytech-fyi/assets/112858669/e76990b3-3cd6-408f-975f-a4cf50cff103)

Let me know what you guys think.